src/CMakeLists.txt

```diff
@@ -112,6 +113,7 @@ TiledArray/pmap/pmap.h
 TiledArray/pmap/replicated_pmap.h
 TiledArray/policies/dense_policy.h
 TiledArray/policies/sparse_policy.h
+TiledArray/SpecialArrays/diagonal_array.h
```

Comment:
change dir name SpecialArrays -> special to match the convention already in place